src/solace_ai_connector/components/general/litellm/litellm_chat_model_with_history.py

```diff
@@ -0,0 +1,106 @@
+"""LiteLLM chat model component with conversation history"""
+
+import time
+
+from .litellm_base import LiteLLMChatModelBase, litellm_info_base
+from ..utils.chat_history_handler import ChatHistoryHandler
+from ....common.log import log
+
+info = litellm_info_base.copy()
+info["class_name"] = "LiteLLMChatModelWithHistory"
+info["description"] = "LiteLLM model handler component with conversation history"
+info["config_parameters"].extend(
+    [
+        {
+            "name": "history_max_turns",
+            "required": False,
+            "description": "Maximum number of conversation turns to keep in history",
+            "default": 10,
+        },
+        {
+            "name": "history_max_time",
+            "required": False,
+            "description": "Maximum time to keep conversation history (in seconds)",
+            "default": 3600,
+        },
+    ]
+)
+
+info["input_schema"]["properties"]["clear_history_but_keep_depth"] = {
+    "type": "integer",
+    "minimum": 0,
+    "description": "Clear history but keep the last N messages. If 0, clear all history. If not set, do not clear history.",
+}
+
+
+class LiteLLMChatModelWithHistory(LiteLLMChatModelBase, ChatHistoryHandler):
+    def __init__(self, **kwargs):
+        super().__init__(info, **kwargs)
+        self.history_max_turns = self.get_config("history_max_turns", 10)
+        self.history_max_time = self.get_config("history_max_time", 3600)
+        self.history_key = f"{self.flow_name}_{self.name}_history"
+
+        # Set up hourly timer for history cleanup
+        self.add_timer(3600000, "history_cleanup", interval_ms=3600000)
+
+    def invoke(self, message, data):
+        session_id = data.get("session_id")
+        if not session_id:
+             raise ValueError("session_id is not provided")
```

Comment:
Are these errors accessible through the ErrorComponent?

Component: https://github.com/SolaceDev/solace-ai-connector/blob/main/src/solace_ai_connector/components/inputs_outputs/error_input.py

Example: https://github.com/SolaceDev/solace-ai-connector/blob/main/examples/error_handler.yaml